### ISSUE TITLE
ts compile error

### ISSUE THREAD

**Issue by gushiaoke** [nojvek/noice-json-rpc#4]
const wssServer = new WebSocket.Server({ port });
const api = new rpc.Server(wssServer).api();

error TS2345: Argument of type 'Server' is not assignable to parameter of type 'LikeSocketServer'.
  Types of property 'clients' are incompatible.
    Type 'Set<WebSocket>' is not assignable to type 'LikeSocket[] | undefined'.
      Type 'Set<WebSocket>' is not assignable to type 'LikeSocket[]'.
        Property 'includes' is missing in type 'Set<WebSocket>'.

**Comment by nojvek** [comment]
That's quite weird.  Any chance you can paste the full snippet you are using. The tests run everyday via a travis cron job and they ran successfully 16 hours ago totally fine.

**Comment by gushiaoke** [comment]
types/ws@0.0.32 index.d.ts
```typescript
// WebSocket Server
export class Server extends events.EventEmitter {
        clients: WebSocket[];
    }
```

types/ws@3.2.1 index.d.ts
```typescript
    // WebSocket Server
    class Server extends events.EventEmitter {
        clients: Set<WebSocket>;
    }
```

noice-json-rpc@1.0.1 noice-json-rpc.d.ts
```typescript
export interface LikeSocketServer {
    clients?: LikeSocket[];
}
```

**Comment by nojvek** [comment]
Looks like Set is the issue. 

I'll move it to ArrayLike and ws 3.2

**Comment by gushiaoke** [comment]
👍

**Comment by nojvek** [comment]
Thanks for reporting. I moved it to Iterable<LikeSocket> which both sets and array are part of.

Published in noice-json-rpc@1.1.1
